src/uid2Sdk.ts

```diff
@@ -219,11 +221,19 @@ export class UID2 {
 
     this._opts = opts;
     this._cookieManager = new UID2CookieManager({ ...opts });
+    this._localStorageManager = new UID2LocalStorageManager();
     this._apiClient = new Uid2ApiClient(opts);
     this._tokenPromiseHandler.registerApiClient(this._apiClient);
-    const identity = this._opts.identity
-      ? this._opts.identity
-      : this._cookieManager.loadIdentityFromCookie();
+
+    let identity;
+    if (this._opts.identity) {
+      identity = this._opts.identity;
+    } else {
+      const localStorageIdentity = this._localStorageManager.loadIdentityFromLocalStorage();
+      const cookieIdentity = this._cookieManager.loadIdentityFromCookie();
+      const shouldUseCookie = cookieIdentity && (!localStorageIdentity || cookieIdentity.identity_expires > localStorageIdentity.identity_expires);
```

Comment:
Hopefully this handles the case we talked about @lionell-pack-ttd, where the cookie is newer than the value in localStorage, so we should use the cookie instead.